virttest/env_process.py

```diff
@@ -314,6 +314,14 @@ def check_image(test, params, image_name, vm_process_status=None):
                               " Skip the image check.")
                 check_image_flag = False
 
+    # Save the potential bad image when the test is not passed.
+    # It should before image check.
+    if (params["test_passed"] == "False" and
+            params.get("save_image_on_error", "no") == "yes"):
+        if vm_process_status == "dead":
+            image.backup_image(params, data_dir.get_data_dir(),
```

Comment:
Have you considered using `test.outputdir`? I think that would be better location to store the test-dependent data. I don't know how about you but I'm only archiving test results and recreating the host, therefor this feature would be useless for me.

Reply:
I would think put all images together would be easier to manage.

As this is only for debugging purpose, as long as the error image exist it is useful, I don't see any reason when it is not in the log dir, it becomes "useless".Just like other images created in `data_dir.get_data_dir()`, does not make them useless when someone use it to debug. 

What's more, put it into log dir makes log dir to large, which it should be as small as possible.

Reply:
That's why I wrote "Have you considered". This is not a strong request. I'm more inclined towards using outputdir, because that is what test produces and in automated fashion that is what usually gets stored. The avocado-datadir is variable, it gets refreshed/erased/recreated frequently. The size is again the question of what people want. By default I'd disable this feature as in most cases one does not care. But when one executes it with this feature enabled, they should expect large file attached to results.

Anyway, we don't follow those strictly so either way is fine by me.